src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildItemBase.cs

```diff
@@ -15,7 +17,15 @@ internal abstract class MSBuildItemBase : IMSBuildItem
         public virtual string Identity { get; }
 
         /// <inheritdoc cref="IMSBuildItem.Properties" />
-        public virtual IReadOnlyList<string> Properties { get; }
+        public IReadOnlyList<string> Properties
+        {
+            get
+            {
+                Debug.Fail("This property should not be accessed.  Calculating the names of the item metadata allocates a new list and can cause performance issues.  Use GetProperty(string) instead and check if the value is null or whitespace.");
```

Comment:
I feel like Debug checks like that are not super helpful, since they rely on people debugging restore, and it's not something that's done that frequently.

Can we throw and mark the method as Obsolete?

Reply:
What I kind of expected is that unit tests or restores with locally built bits would fail if any future code path accessed this property.  I'll mark it as obsolete and throw instead.  This class is internal so at least we know no one is already using it.